src/main/java/com/sfjs/security/ApiKeyFilter.java

```diff
@@ -0,0 +1,34 @@
+package com.sfjs.security;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+@Component
+public class ApiKeyFilter extends OncePerRequestFilter {
+
+  private static final String AUTH_TOKEN_HEADER_NAME = "X-API-KEY";
+
+  @Value("${api.key}")
+  private String API_KEY;
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+    String apiKey = request.getHeader(AUTH_TOKEN_HEADER_NAME);
+
+    if (apiKey == null || !apiKey.equals(API_KEY)) {
```

Comment:
My Java is a bit rusty, would you mind briefly explaining to me what this condition means?
```java
!apiKey.equals(API_KEY)
```

Reply:
Yes, the API_KEY is a unique value only we know.  It is only expressed in the environment variables in Vercel and Render.  It is not in github.  So there is no way anyone can get it.